docs/user_guide/plugin_development/api-reference.md

```diff
@@ -0,0 +1,286 @@
+---
+title: Plugin API Reference
+summary: "Complete API reference for Beekeeper Studio plugin development."
+icon: material/api
+---
+
+# Plugin API Reference
+
+!!! warning "Beta Feature"
+    The plugin system is currently in beta. APIs and functionality may change in future releases. Please provide feedback and report issues to help us improve the system.
+
+The Beekeeper Studio Plugin API is accessible through the `@beekeeperstudio/plugin` package. The API provides three main functions for communication between your plugin and the main application.
+
+## Installation
+
+Install the plugin package from GitHub:
+
+```bash
+npm install github:beekeeper-studio/plugin
+```
+
+## Import API Functions
+
+```javascript
+import { request, notify, addNotificationListener } from '@beekeeperstudio/plugin';
+```
+
+## How It Works
```

Comment:
Can we add the example plugin (the one you have screenshots of) link here, so folks can look at the real code?